packages/ui/src/ui/containers/AuthenticationTokensModal/AuthenticationTokensModal.tsx

```diff
@@ -0,0 +1,397 @@
+    const [token, setToken] = useState<string>();
+
+    const handleSubmit = (form: FormApi<FormData>) => {
+        const {description} = form.getState().values;
+
+        setError(null);
+
+        return dispatch(
+            authenticationTokensCreateToken({
+                description,
+                credentials,
+            }),
+        )
+            .then((token) => {
+                setToken(token);
+
+                return dispatch(authenticationTokensGetList(credentials))
+                    .then(() => undefined)
+                    .catch(() => undefined);
+            })
+            .catch((error) => {
+                setError(error);
+            });
+    };
+
+    if (token) {
+        return (
+            <div className={block('tokens')}>
+                <h2>Copy token value</h2>
+                <Alert
+                    theme="warning"
+                    message="Please save the token value. It is impossible to get the token value after closing the dialog"
+                />
+                <br />
+                <Alert
+                    message={token}
+                    layout="horizontal"
+                    actions={<ClipboardButton text={token} />}
+                />
+                <div className={block('tokens-action')}>
+                    <Button
+                        size="l"
+                        view="action"
+                        onClick={() => {
+                            setSection(ViewSection.tokens);
+                        }}
+                    >
+                        Close
+                    </Button>
+                </div>
+            </div>
+        );
+    }
+
+    return (
+        <YTDFDialog<FormData>
+            headerProps={{
+                title: 'Generate token',
+            }}
+            pristineSubmittable
+            modal={false}
+            visible={true}
+            initialValues={{}}
+            onAdd={handleSubmit}
+            fields={[
+                {
+                    name: 'description',
+                    type: 'textarea',
+                    required: true,
+                    caption: 'Description',
+                },
+                {
+                    name: 'error-block',
+                    type: 'block',
+                    extras: {
+                        children: error && (
+                            <Error message="Error during toke generating" error={error} />
+                        ),
+                    },
+                },
+            ]}
+            footerProps={{
+                propsButtonCancel: {
+                    onClick: () => {
+                        setSection(ViewSection.tokens);
+                    },
+                },
+            }}
+        />
+    );
+};
+
+type DefaultProps = {
+    setSection: React.Dispatch<React.SetStateAction<ViewSection>>;
+};
+
+const AuthenticationPasswordSection: FC<DefaultProps> = ({setSection}) => {
```

Comment:
The component should not know anything about sections. I hope we will get rid of the component cause we have oauth, see comment for `authenticationTokensSetCredentials`.